lib/rrule/dtstart.rs

```diff
@@ -34,7 +34,10 @@ impl DtStart {
     let mut parameters = Parameters::new();
 
     if let Some(tzid) = self.tzid {
-      parameters.insert("TZID".to_string(), tzid.to_string());
+      // UTC datetimes MUST NOT contain a TZID
+      if tzid != chrono_tz::Tz::UTC {
```

Comment:
```suggestion
      if !datetime.utc() {
```

TZID should not be applied if datetime is in UTC

Reply:
Hi, sorry for silence. I've been on vacation.

Looks like the validation issue you mentioned happen because of the missing VTIMEZONE block. For example, if I add it, it works well:

```icalendar
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//YourOrg//YourApp//EN
BEGIN:VTIMEZONE
TZID:UTC
BEGIN:STANDARD
DTSTART:19981025T020000
TZOFFSETFROM:-0400
TZOFFSETTO:-0500
TZNAME:EST
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
UID:12345678-90ab-cdef-1234-567890abcdef@example.com
DTSTAMP:20240718T120000Z
DTSTART;TZID=UTC:20250101T000000
DURATION:PT1H
SUMMARY:Team Sync Meeting
RRULE:FREQ=WEEKLY;COUNT=10
END:VEVENT
END:VCALENDAR
```

But if I use utc for datetime, it will fail:


```icalendar
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//YourOrg//YourApp//EN
BEGIN:VTIMEZONE
TZID:UTC
BEGIN:STANDARD
DTSTART:19981025T020000
TZOFFSETFROM:-0400
TZOFFSETTO:-0500
TZNAME:EST
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
UID:12345678-90ab-cdef-1234-567890abcdef@example.com
DTSTAMP:20240718T120000Z
DTSTART;TZID=UTC:20250101T000000Z
DURATION:PT1H
SUMMARY:Team Sync Meeting
RRULE:FREQ=WEEKLY;COUNT=10
END:VEVENT
END:VCALENDAR
```

> Invalid DTSTART value, The 'TZID' property parameter MUST NOT be applied to DATE-TIME properties whose time values are specified in UTC. [near line # 13](javascript:void(0);)
Reference: [3.3.5. Date-Time](http://icalendar.org/iCalendar-RFC-5545/3-3-5-date-time.html)

So, looks like TZID + local datetime is a valid case

Reply:
ok ! I've updated the PR to remove the changes made to the other tests. 👍